x-pack/plugins/uptime/server/lib/helper/get_filtered_query.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, set } from 'lodash';
+
+export const getFilteredQuery = (
+  dateRangeStart: string,
+  dateRangeEnd: string,
+  filters?: string | null
+) => {
+  let filtersObj;
+  // TODO: handle bad JSON gracefully
+  filtersObj = filters ? JSON.parse(filters) : undefined;
+  if (get(filtersObj, 'bool.must', undefined)) {
+    const userFilters = get(filtersObj, 'bool.must', []);
+    delete filtersObj.bool.must;
+    filtersObj.bool.filter = [...userFilters];
+  }
+  const query = { ...filtersObj };
+  const rangeSection = {
+    range: {
+      '@timestamp': {
+        gte: dateRangeStart,
+        lte: dateRangeEnd,
+      },
+    },
+  };
+  if (get(query, 'bool.filter', undefined)) {
+    query.bool.filter.push({
```

Comment:
```diff
-query.bool.filter.push({
-  ...rangeSection,
-});
+query.bool.filter.push(rangeSection);
```

Reply:
Yeah that's really stupid. Thanks!
Resolved https://github.com/elastic/kibana/pull/29374/commits/8178a47ea9e7566f4b4b1e3377ffe2217398009f.